### PR TITLE
Feature/continueonnil

### PR DESCRIPTION
**Pull request by nhooyr**

### CODE REVIEW

File: retry.go [hunk 1]
```diff
@@ -37,24 +37,27 @@ func New(sleep time.Duration) *Retry {
 	return r
 }
 
+// ContinueOnNil makes the retry continue even if the run function
+// returns nil. You will need to set explicit conditions for the
+// retry to exit.
+func (r *Retry) ContinueOnNil() *Retry {
+	// The first post condition is the one that returns false
+	// if the error is nil so we need to remove it.
+	r.postConditions = r.postConditions[1:]
```
Comment (ammario) [inline retry.go]: This is pretty janky, but I trust that you've considered alternatives.

File: retry.go [hunk 2]
```diff
@@ -124,14 +127,23 @@ func (r *Retry) postCheck(err error) bool {
...
+// Deprecated: Use MaxAttempts.
 func (r *Retry) Attempts(maxAttempts int) *Retry {
+	return r.MaxAttempts(maxAttempts)
+}
+
+// MaxAttempts sets the maximum amount of retry attempts
+// before the current error is returned.
+// If maxAttempts is 0, then r.Run() will return a nil
+// error on any call.
+func (r *Retry) MaxAttempts(maxAttempts int) *Retry {
```
Comment (ammario) [inline retry.go]: I'd prefer keeping `Attempts` until vgo proliferates. It's not that bad of a name.

Comment (ammario) [reply retry.go]: I think this API has a good potential of becoming standard in the Go community. I feel like a lot of deprecation will scare off developers.

File: retry.go (hunk 1 above)
Comment (nhooyr) [reply retry.go]: Did not actually, I agree its pretty janky. I just did it like that to save time. I'll fix.

File: retry.go (hunk 2 above)
Comment (nhooyr) [reply retry.go]: Fair enough.

**Comment by nhooyr** [comment]
DAMMIT I COMMITED A FIXUP.

**Pull request merged**